wolfcrypt/src/aes.c

```diff
@@ -12433,5 +12411,27 @@ int wc_AesSivDecrypt(const byte* key, word32 keySz, const byte* assoc,
 
 #endif /* WOLFSSL_AES_SIV */
 
+#ifdef WOLFSSL_SE050
+/**
+ * Use software crypto instead of hardware cryptography. Only supported on
+ * NXP SE050 currently.
+ *
+ * aes      pointer to initialized Aes struct
+ * enabled  0 to use hardware crypto, 1 to use software crypto
+ *
+ * Returns 0 on success, negative on error
+ */
+int wc_AesUseSoftware(Aes* aes, byte enabled)
```

Comment:
Can we just use the `devId` to gate this? Init with `INVALID_DEVID` use software or != INVALID_DEVID to use SE050 HW?

Reply:
I needed to switch SE050 to use a new `word32 keyId` in the Aes struct instead of our `int devId`.  SE050 key ID's can go up to 0xFFFFFFFF (max size of an "unsigned int").  The "int" type of devId is too small to hold the upper top range of key ID's, several of which represent default NXP credentials.

I didn't really like the idea of adding `wc_AesUseSoftware`, but when wolfSSL is on the host side, the HostCrypto layer needs to force wolfSSL to use software crypto, even for things like key generation.  If switching just on `keyId`, which would start at 0 for word32, an API like `wc_MakeRsaKey()` would auto-pick a new key  ID and try to generate it in SE050.  That would fail since SE050 either isn't authenticated or available at that point on the HostCrypto side.

Only other alternative I saw was not having this API, and make the application/HostCrypto code explicitly set `aes->useSWCrypt = 1` directly.  That would work, and maybe be OK since we control the HostCrypto code calling this.  Just had leaned away from this approach with direct structure access.  Maybe better to go back to this route?

Reply:
I removed the `*UseSoftware()` functions and modified SE050 HostCrypto to directly set the struct member.